modules/govuk_mtail/manifests/init.pp

```diff
@@ -0,0 +1,76 @@
+# == Class: govuk_mtail
+#
+# This class installs and sets-up Mtail
+#
+# === Parameters
+#
+# [*manage_repo_class*]
+#   Whether to use a separate repository to install Statsd
+#   Default: false (use 'govuk_ppa' repository)
+#
+# [*logs*]
+#   List of files to monitor
+#
+# [*enabled*]
+#   Enable Mtail service. (default true)
+#
+# [*port*]
+#   HTTP port to listen on. (default "3903")
+#
+# [*collectd_socketpath*]
+#   Path to collectd unixsock to write metrics to.
+#
+# [*graphite_host_port*]
+#   Host:port to graphite carbon server to write metrics to.
+#
+# [*statsd_hostport*]
+#   Host:port to statsd server to write metrics to.
+#
+# [*metric_push_interval*]
+#   Interval between metric pushes, in seconds (default 60)
+#
+# [*extra_args*]
+#   Mtail program extra arguments (default -log_dir /var/log/mtail)
+#
+class govuk_mtail(
+  $logs = [],
```

Comment:
we could add a `validate_array` function for this

Reply:
I removed the parameter because we are doing a lookup with the `hiera_array` function. If we don't want to do a recursive merge, we should remove that function and put the variable back here

Reply:
I think it's clearer to understand if we use a parameter rather than `hiera_array` lookup as it makes it more portable as a class. I think we should also do checks incase it's an empty value, and avoid adding mtail if there are no logs to parse.

Reply:
I thought in doing a merge so each application can declare an individual log in its Hieradata file without having to use a common variable in Hieradata for the entire infrastructure. The `hiera_array` function is already passing a default so it'll never pass an empty variable to the mtail class.

What you suggest is in line with the way in which we are doing things right now so I'll change this code, but maybe when we start moving to version 5 we could consider other options?